idseq_dag/steps/run_srst2.py

```diff
@@ -0,0 +1,140 @@
+            command.execute(f"cp empty_file.txt {self.output_files_local()[3]}")
+            command.execute(f"cp empty_file.txt {self.output_files_local()[4]}")
+        else:
+            # Post processing of amr data
+            # TODO: Handle case where local output file is named the same as that in dag and mv fails with "same file" error
+            command.execute(f"mv {os.path.join(self.output_dir_local, 'output__fullgenes__ARGannot_r2__results.txt')} {self.output_files_local()[2]}")
+            PipelineStepRunSRST2.generate_processed_amr_full_results(self.output_files_local()[2], self.output_files_local()[3])
+            PipelineStepRunSRST2.generate_processed_amr_summary_results(self.output_files_local()[2], self.output_files_local()[4])
+
+    # Inherited method
+    def count_reads(self):
+        pass
+   
+    @staticmethod 
+    def generate_processed_amr_full_results(amr_raw_path, results_path):
+        amr_processed_results = PipelineStepRunSRST2._process_amr_results(amr_raw_path)
+        amr_processed_results.to_csv(
+            'amr_processed_results.csv',
+            mode='w',
+            index=False,
+            encoding='utf-8')
+        command.execute(f"mv amr_processed_results.csv {results_path}")        
+    
+    @staticmethod
+    def generate_processed_amr_summary_results(amr_raw_path, results_path):
+        amr_summary_results = PipelineStepRunSRST2._summarize_amr_gene_families(amr_raw_path)
+        amr_summary_results.to_csv(
+            'amr_summary_results.csv',
+            mode='w',
+            index=False,
+            encoding='utf-8')
+        command.execute(f"mv amr_summary_results.csv {results_path}")
+
+    @staticmethod
+    def _get_pre_proc_amr_results(amr_raw_path):
+        """ Reads in raw amr results file outputted by srst2, and does initial processing of marking gene family,
+        + adding in blank columns for relevant metrics (total_genes_hit, total_coverage, total_depth)."""
+        amr_results = pd.read_csv(amr_raw_path, delimiter="\t")
+        # Parse out gene family as substring after '_', e.g. Aph_AGly's gene family would be AGly
+        amr_results['gene_family'] = amr_results.apply(lambda row: row.gene[row.gene.index('_')+1:], axis=1)
+        # Add empty columns for later processing
+        amr_results['total_genes_hit'], amr_results['total_coverage'], amr_results['total_depth'] = (
+            [amr_results.apply(lambda _: '', axis=1)]) * 3
+        return amr_results
+
+    @staticmethod
+    def _summarize_amr_gene_families(amr_raw_path):
+        """Returns a gene family-level summary of total_genes_hit, total_coverage, and total_depth."""
+        amr_results = PipelineStepRunSRST2._get_pre_proc_amr_results(amr_raw_path)
+        amr_results_by_gene_fam = amr_results.groupby(['gene_family'])
+        total_genes_hit, total_coverage, total_depth = (amr_results_by_gene_fam.size().reset_index(name='total_genes_hit'),
```

Comment:
you found "groupby"... now check out aggregation (`agg`) to complete the right way of doing this :)
https://pandas.pydata.org/pandas-docs/version/0.23/generated/pandas.core.groupby.DataFrameGroupBy.agg.html
This removes the need to do the merges at the end of this function.